src/main/java/dev/rollczi/antilegs/utils/InventoryUtils.java

```diff
@@ -27,17 +27,12 @@ public static boolean hasFreeSlotFor(Inventory inventory, ItemStack item) {
         }
 
         for (ItemStack itemInv : inventory.getContents()) {
-            if (itemInv == null) {
-                continue;
-            }
 
-            if (!itemInv.isSimilar(item)) {
+            if (itemInv == null || !itemInv.isSimilar(item)) {
                 continue;
             }
 
-            if (itemInv.getMaxStackSize() > itemInv.getAmount()) {
-                return true;
-            }
+            return itemInv.getMaxStackSize() > itemInv.getAmount();
```

Comment:
to zmienia logikę pętli, `return` miało zwracać tutaj tylko true albo kontynuować pętlę